addon/-private/validated-changeset.ts

```diff
@@ -851,14 +857,32 @@ export class BufferedChangeset implements IChangeset {
       }
 
       if (result && isObject(result)) {
-        return normalizeObject(result);
-      }
+        result = normalizeObject(result);
+         let content: Content = this[CONTENT];
+
+        // Merge the content with the changes to have a complete object for a nested property.
+        // Given a object with nested property and multiple properties inside of it, if the 
+        // requested key is the top most nested property and we have changes in of the properties, we need to 
+        // merge the original model data with the changes to have the complete object.
+        // eg. model = { user: { name: 'not changed', email: 'changed@prop.com'} }
+        if (isObject(content[baseKey]) && !isEmberDataObject(content[baseKey])) {
+          let data: { [key: string]: any } = {}
```

Comment:
`let data: Record<string, any> = {}`

is another option

Reply:
Changed.